src/mdacli/cli.py

```diff
@@ -541,27 +548,25 @@ def main():
         # Ignore all warnings if not in debug mode
         warnings.filterwarnings("ignore")
 
-    # Execute the main client interface.
-    try:
-        analysis_callable = args.analysis_callable
-
-        # Get the correct ArgumentParser instance from all subparsers
-        # `[0]` selects the first subparser where our analysises live in.
-        ap_sup = ap._subparsers._group_actions[0].choices[
-            analysis_callable.__name__]
-        arg_grouped_dict = split_argparse_into_groups(ap_sup, args)
-
-        # Optional parameters may not exist
-        arg_grouped_dict.setdefault("Optional Parameters", {})
-
-        run_analsis(analysis_callable,
-                    arg_grouped_dict["Universe Parameters"],
-                    arg_grouped_dict["Mandatory Parameters"],
-                    arg_grouped_dict["Optional Parameters"],
-                    arg_grouped_dict["Analysis Run Parameters"],
-                    arg_grouped_dict["Output Parameters"])
-    except Exception as e:
-        if args.debug:
-            traceback.print_exc()
-        else:
-            sys.exit(Emphasise.error(f"Error: {e}"))
+    with setup_logging(logfile=args.logfile, debug=args.debug):
```

Comment:
I never did logging as a `context`, but it looks very pythonic. I think it serves well as long there is no need to change the log configuration anywhere else during run time, but that doesn't seem to be the case anyway because `mdacli` is pretty much straightforward (conceptually). :clap: :+1:

Reply:
Thanks!